### ISSUE TITLE
Pressing end doesn't scroll all the way to the end of a line

### ISSUE THREAD

**Issue by TheV360** [processing/p5.js-web-editor#745]
#### Nature of issue?

- [x] Found a bug
- [ ] Existing feature enhancement
- [ ] New feature request

#### Details about the bug: 

- Web browser and version: Firefox 63
- Operating System: Windows 10 1803
- Steps to reproduce this:
1. Make file with a very long line of code that doesn't fit on the screen.
```js
function heck() {
  extremelyLongFunctionCall(helloWorld, xCoordinate, yCoordinate, otherVariableName, runningOutOfCreativeVariableNames, test, justRanOutOfVariableNames);
}
```
2. On the long line of code, press End.
3. You'll see that while the cursor moved to the end of the line correctly, the scrollbar didn't catch up to it, and you can't easily see what's at the end of the line without dragging the scrollbar.

**Comment by bhawesh96** [comment]
As far as I could figure out, it has to do something related to the `.CodeMirror-scroll` class in `_codemirror.scss` file.

**Comment by catarak** [comment]
@bhawesh96 thanks for looking into that!

**Comment by siddharthshah3030** [comment]
@catarak 
I would like to solve this issue but can't figure out how to approach
I tried modifying the file mentioned by @bhawesh96 
and tried changes in CSS attributes
but it didn't seem to work for me
I'm using chrome (Manjaro Linux)

**Comment by LakshSingla** [comment]
> thanks for reporting! i've changed the styling of CodeMirror a bunch, so I wonder if it has to do with that.

Can you tell some key areas where you have modified the code so that I can take a look at it ?

**Comment by catarak** [comment]
if you look in https://github.com/processing/p5.js-web-editor/blob/master/client/styles/vendors/_codemirror.scss, you can see the basic CodeMirror styles, and in https://github.com/processing/p5.js-web-editor/blob/master/client/styles/components/_editor.scss, some of the custom styles.

**Comment by bmoren** [comment]
I'd also like to add that there is an issue while typing that I believe is related to this one where the scroll bar is not caught up to the overflowing text:

![ezgif com-optimize](https://user-images.githubusercontent.com/1385996/53663126-8342d900-3c2a-11e9-8e9d-ece3b6bb2fae.gif)

**Comment by catarak** [comment]
@bmoren that does look like the same issue!

**Comment by bmoren** [comment]
related, flexbox issue: 

https://github.com/codemirror/CodeMirror/issues/4142

**Comment by AnkurIngale** [comment]
I'll take this one.

**Comment by catarak** [comment]
the CodeMirror editor is in a flexbox container, so it could the same issue that @bmoren linked to.

**Comment by shakti97** [comment]
I think we can close this issue as I can see we are now not using the horizontal scroll with the editor and vertical scroll is working fine

**Comment by aldrinjenson** [comment]
Yes, this issue is no more a problem. Now the code wraps down to the next line. Please close the issue.

**Comment by TheV360** [comment]
The previous two comments are saying that this is fixed, but I went into settings, disabled Word Wrap, and immediately got the issue to happen again.

![2020-03-22_21-21-14](https://user-images.githubusercontent.com/25177071/77270739-2970df00-6c83-11ea-94c7-24ae9036f6f1.gif)

**Comment by sagar-joshi** [comment]
The actual issue was with codemirror and now it has been fixed in codemirror.
https://github.com/codemirror/CodeMirror/issues/6436

**Comment by catarak** [comment]
> The actual issue was with codemirror and now it has been fixed in codemirror.

Wow really?! That's exciting. I'll leave this open until there's a new Codemirror release and this can be updated!

**Comment by sagar-joshi** [comment]
> Wow really?!

I had tested with 'fixedGutter : true' and scrolling was working perfect. The problem was with 'fixedGutter : false' (and currently fixedGutter is set to false in p5js web editor). Now it is fixed in codemirror. So it should work.

**Comment by sagar-joshi** [comment]
@catarak codemirror has released new v5.58.2

**Comment by catarak** [comment]
@sagar-joshi thank you! I'll make a PR that fixes this 😄
